autorecsys/pipeline/preprocessor.py

```diff
@@ -78,6 +103,216 @@ def preprocessing(self, **kwargs):
         raise NotImplementedError
 
 
+class AvazuPreprocessor(BaseCTRPreprocessor):
+
+    def __init__(self, dataset_path, save_path):
+        super(AvazuPreprocessor, self).__init__(dataset_path=dataset_path, )
+        self.save_path = save_path
+        self.categ_filter = 10
+
+        self.columns_names = ["id", "click", "hour", "C1", "banner_pos", "site_id", "site_domain", "site_category",
+                              "app_id", "app_domain", "app_category", "device_id", "device_ip", "device_model",
+                              "device_type", "device_conn_type", "C14", "C15", "C16", "C17", "C18", "C19", "C20", "C21"]
+        self.used_columns_names = self.columns_names
+        self.dtype_dict = {n: np.float32 for n in self.used_columns_names}
+
+        self._load_data()
+
+    def _load_data(self):
+        raise NotImplementedError
+
+    def preprocessing(self, test_size, random_state):
+        raise NotImplementedError
+
+
+class CriteoPreprocessor(BaseCTRPreprocessor):
+
+    def __init__(self):
+        # Step 1: Set attributes used when initializing the preprocessor object.
+        # dataset_path = "./examples/datasets/criteo_full/train.txt"
+        dataset_path = "./examples/datasets/criteo_sample_10000/train_examples.txt"
```

Comment:
can we add a path argument in the init and put this path as default?

Reply:
Agreed.